sqlglot/dialects/risingwave.py

```diff
@@ -1,6 +1,501 @@
+from __future__ import annotations
 from sqlglot.dialects.postgres import Postgres
+from sqlglot.tokens import TokenType
+from collections import defaultdict
+import typing as t
+
+from sqlglot import exp
 
 
 class RisingWave(Postgres):
+    class Tokenizer(Postgres.Tokenizer):
+        KEYWORDS = {
+            **Postgres.Tokenizer.KEYWORDS,
+            "SINK": TokenType.SINK,
+            "SOURCE": TokenType.SOURCE,
+        }
+
+    class Parser(Postgres.Parser):
+        DB_CREATABLES = {
+            *Postgres.Parser.DB_CREATABLES,
+            TokenType.SINK,
+            TokenType.SOURCE,
+        }
+        CREATABLES = {
+            *Postgres.Parser.CREATABLES,
+            *DB_CREATABLES,
+        }
+
+        ID_VAR_TOKENS = {
+            *Postgres.Parser.ID_VAR_TOKENS,
+            *CREATABLES,
+        }
+
+        def _parse_table_hints(self) -> t.Optional[t.List[exp.Expression]]:
+            # There is no hint in risingwave.
+            # Do nothing here to avoid WITH keywords conflict in CREATE SINK statement.
+            return None
+
+        def _parse_include(self) -> t.Optional[exp.Expression]:
```

Comment:
This method looks too complicated, can you explain what the different args of `Include` represent? I think we can use `_parse_var_from_options` for `column_type`, for example. The rest can probably be cleaned up as well.

Reply:
From risingwave `INCLUDE` clause, it should be used as `INCLUDE { header | key | offset | partition | timestamp } [AS <column_name>]`. One special case is `INCLUDE header 'foo' varchar AS myheader` where `foo` and `varchar` could be provided optionally here.
And it is parsed by risingwave here: https://github.com/risingwavelabs/risingwave/blob/main/src/sqlparser/src/parser.rs#L2607.
I then generally follow the logic. I try to see `_parse_var_from_options`, but I don't think it will work as the configuration is not fixed.